survey-sdk/src/main/java/co/nimblehq/survey/sdk/entity/SurveyEntity.kt

```diff
@@ -0,0 +1,46 @@
+package co.nimblehq.survey.sdk.entity
+
+import com.squareup.moshi.Json
+import moe.banana.jsonapi2.HasMany
+import moe.banana.jsonapi2.JsonApi
+import moe.banana.jsonapi2.Resource
+
+@JsonApi(type = "survey")
+class SurveyEntity : Resource() {
+    var title: String? = ""
+    var description: String = ""
+
+    @field:Json(name = "thank_email_above_threshold")
+    var thankEmailAboveThreshold: String = ""
+
+    @field:Json(name = "thank_email_below_threshold")
+    var thankEmailBelowThreshold: String? = ""
+
+    @field:Json(name = "is_active")
+    var isActive: Boolean? = false
+
+    @field:Json(name = "cover_image_url")
+    var coverImageURL: String? = ""
+
+    @field:Json(name = "created_at")
+    var createdAt: String? = ""
+
+    @field:Json(name = "active_at")
+    var activeAt: String? = ""
+
+    @field:Json(name = "survey_type")
+    var surveyType: String? = ""
+
+    var questions: HasMany<QuestionEntity>? = null
+
+    fun getQuestionList(): List<QuestionEntity>? {
+        val resources = questions?.get()
+        return resources?.map { resource ->
+            QuestionEntity().apply {
+                id = resource.id
+                type = resource.type
+            }
+        }
```

Comment:
Same here

Reply:
@hoangnguyen92dn Not for this one. If the `question` only has `id` and `type`, when using with `document`, we will get the list with all `null` elements

<img width="991" alt="Screen Shot 2021-09-15 at 2 42 38 PM" src="https://user-images.githubusercontent.com/88996251/133392150-b0eb3735-ad30-44e2-8df7-8f043e265edf.png">

<img width="991" alt="Screen Shot 2021-09-15 at 2 42 38 PM" src="https://user-images.githubusercontent.com/88996251/133392233-7e10f21c-b98b-4746-8dd6-b8f3d85475b9.png">

Reply:
@doannimble I am not sure about your input test data. If you do it right, the library will map the objects for you by using `id` and `type` instead of doing manually.

Reply:
@hoangnguyen92dn you can check the `DataTest.kt` for reference. 
I'm trying to digging deeply inside the lib , and see the function : 
`public List<T> get(Document document, T defaultValue) {
        List<T> collector = new ArrayList<>(linkedResources.size());
        for (ResourceIdentifier resourceId : linkedResources) {
            T obj = document.find(resourceId);
            collector.add(obj == null ? defaultValue : obj);
        }
        return collector;
    }`

`linkedResources` has data (id, type), but then calling document.find, they cannot find as 
`public <T extends Resource> T find(ResourceIdentifier resourceIdentifier) {
        return (T) included.get(resourceIdentifier);
    }`
 You can see , it find from `included`.
But regarding to the response of the list, there is no `included` : https://documenter.getpostman.com/view/11835486/T1LJjTqj#813ee4aa-2f48-44e3-ad2b-d65f6babb02a

Reply:
> But regarding to the response of the list, there is no included

![image](https://user-images.githubusercontent.com/6950766/133546601-c17f1cb2-93fb-46a6-a1c2-43e6a1938165.png)

@doannimble Please double-check our input test data. I see there is `included` in the details mock response. And the list contains 2 mappings for questions but in only 1 object for the first `id` (I marked 1 and 2 from screenshot)

![image](https://user-images.githubusercontent.com/6950766/133546306-fa186f54-a39c-4501-b198-371ed324b1c1.png)

Reply:
@hoangnguyen92dn Agree with this, as we dont need to care about the redundant data from api. So I've update the test case along with getQuestionList function 🙏

Reply:
@doannimble As discussed, I'd recommend having `Survey` and `SurveyDetails` for usage instead of combine in a single object, because the `questions` field is unnecessary from the survey list, we only use it for the details page.

Reply:
@hoangnguyen92dn I suggest to consider about it later, as it has others inconvenience like no reusable, must create more adapter and more class.